build/main.rs

```diff
@@ -18,7 +18,17 @@ fn main() {
 }
 
 fn generate_gl_bindings<W>(dest: &mut W) where W: Write {
-    let gl_registry = Registry::new(
+    let gl_registry_4_6 = Registry::new(
```

Comment:
Can you explain why creating an entirely new registry is needed? Can't it just be integrated into the gl_registry?

Reply:
Not sure if it's valid to do it in the 4.5 registry.
[This site](https://www.khronos.org/opengl/wiki/SPIR-V) says that it's available in 4.6 core, or with the  `ARB_gl_spirv` extension in earlier versions.
My question for you is: What does the version of a registry mean? Does the version of a registry mean that the extensions specified for it will only be generated when the installed OpenGL version is below the version of the registry?
IOW, how does this translate to `Registry` instantiation:

![image](https://user-images.githubusercontent.com/535593/101988308-154d9180-3c99-11eb-9ebc-8430c5700a7c.png)

Maybe I should also use [`Profile::Core`](https://docs.rs/gl_generator/0.14.0/gl_generator/enum.Profile.html#variant.Core) instead of `Profile::Compatibility`?

Reply:
> My question for you is: What does the version of a registry mean? Does the version of a registry mean that the extensions specified for it will only be generated when the installed OpenGL version is below the version of the registry?

Note that this is in build/main.rs so the code runs at compile time, not runtime. At that point, there is no concept of installed opengl version.

The request for using a specific opengl version is handled in a different part of the code, as is checking for availability of an extension.

Second, I dont know exactly what the minimum opengl version is that glium supports, but it's certainly not 4.5. I don't think  you need two separate registries here, just add the extension and maybe increase the version number in the existing registry. That being said I'm not 100% certain what this number does. gl-generator docs dont show anything, maybe look at the source code of the crate?

Reply:
Ah yeah, I understood it now:
When I only use version `(4, 5)`, then `gl.SpecializeShader` is not generated (because it's part of core OpenGL 4.6), and when I use `(4, 6)` but don't have `"GL_ARB_gl_spirv"` in the extension list, then `gl.SpecializeShaderARB` is not generated.
So I need to have both, version `(4, 6)` and `"GL_ARB_gl_spirv"` in the extension list, then everything works. I updated it in my branch.